frontend/public/models/yaml-templates.ts

```diff
@@ -809,3 +810,20 @@ spec:
     kind: MachineSet
     name: worker
 `);
+
+const pluginTemplates = ImmutableMap<GroupVersionKind, ImmutableMap<string, string>>()
+  .withMutations(map => {
+    plugins.registry.getYamlTemplates().forEach(yt => {
+      const modelRef = referenceForModel(yt.properties.model);
+      const templateName = yt.properties.templateName || 'default';
+
+      if (!baseTemplates.hasIn([modelRef, templateName])) {
+        map.setIn([modelRef, templateName], yt.properties.template);
+      } else {
+        // eslint-disable-next-line no-console
+        console.warn(`attempt to redefine YAML template ${templateName} for model ${modelRef}`);
```

Comment:
What about throwing and `Error`? IMO, this conflict should be avoided.

Reply:
This code gets referenced & evaluated very early during Console startup.

Consequently, throwing an `Error` here will effectively break the whole application - just a blank page with error stack trace shown in the browser console. This is not the behavior we want.

In good browsers, `console.warn` and `console.error` provide you with a link to source code (based on source maps) if you want to investigate the issue.

Reply:
This should be caught at build time.
@vojtechszocs and I talked about fixing this situation by creating unit tests to avoid duplicates and therefore can avoid this type of runtime validation.